mlir/test/Dialect/Linalg/decompose-named-ops.mlir

```diff
@@ -0,0 +1,157 @@
+// RUN: mlir-opt %s -split-input-file -linalg-decompose-named-ops | FileCheck %s
+// RUN: mlir-opt %s -split-input-file -linalg-decompose-named-ops -linalg-generalize-named-ops | FileCheck %s --check-prefix=GENERALIZECHECK
```

Comment:
[nit] With more than one CHECK prefix, you can drop `CHECK` (which is just noise IMHO), and leverage the prefixes to better document what's tested:
```
// RUN: mlir-opt %s -split-input-file -linalg-decompose-named-ops | FileCheck %s -check-prefix=DECOMPOSED
// RUN: mlir-opt %s -split-input-file -linalg-decompose-named-ops -linalg-generalize-named-ops | FileCheck %s --check-prefix=GENERALIZED
```

I am also making the suggestion to make sure that my understanding is correct.